### DISCUSSION TITLE
u8x8 support for ssd1306 128x32 0.91" I2C OLED?

### DISCUSSION THREAD

**Discussion by gnbl** [olikraus/u8g2#1705] [General]
Wow, GitHub kills my input when forcing me to choose a Category!?

Looks like this: https://makerfabs.com/image/cache/makerfabs/0.91%20I2C%20OLED%20128x32-%20Blue/0.91%20I2C%20OLED%20128x32-%20Blue_4-1000x750.png

It seems this type has only SPI support so far in U8x8.
I've hacked u8x8_d_ssd1306_128x64_noname.c using the init data/sequence from https://github.com/adafruit/Adafruit_SSD1306/blob/master/Adafruit_SSD1306.cpp and that seems to work, but there is already a "u8x8_d_ssd1306_128x32_generic" that maybe could be used ...?

**Comment by olikraus** [comment]
Ahm yes, the 128x32 is already supported: 

e.g.: https://github.com/olikraus/u8g2/wiki/u8x8setupcpp#ssd1306-128x32_univision

**Comment by gnbl** [comment]
Right, but SPI only?

**Comment by gnbl** [comment]
As opposed to this one: https://github.com/olikraus/u8g2/wiki/u8x8setupcpp#ssd1306-128x64_noname-1

**Comment by olikraus** [comment]
Usually all combinations are supported by U8g2 / U8x8:

https://github.com/olikraus/u8g2/blob/6b365b63371ff2a1be3971b43b0b19901b8c6712/cppsrc/U8x8lib.h#L1665
https://github.com/olikraus/u8g2/blob/6b365b63371ff2a1be3971b43b0b19901b8c6712/cppsrc/U8x8lib.h#L1671
https://github.com/olikraus/u8g2/blob/6b365b63371ff2a1be3971b43b0b19901b8c6712/cppsrc/U8x8lib.h#L1749

**Comment by gnbl** [comment]
`U8X8_SSD1306_128X32_UNIVISION_SW_I2C ` works, thank you! Missed that in my search.
